### ISSUE TITLE
Replay: RangeError: Invalid Time Value

### ISSUE THREAD

**Issue by alancnet** [getsentry/sentry#47021]
### Environment

SaaS (https://sentry.io/)

### Link

https://jobot.sentry.io/replays/jobot-com-www:0d86c2d0442c4a6ca16aeb7cb3df1af3/?query=&referrer=%2Freplays%2F&statsPeriod=14d&yAxis=count%28%29

### DSN

'https://7b71d2815bd249cf82651234fe64606d@o206778.ingest.sentry.io/6459803'

### Steps to Reproduce

Load a replay.

### Expected Result

The replay loads.

### Actual Result

The replay disappears, replaced by "There was an error loading a component" with "RangeError: Invalid time value" in the console.

![image](https://user-images.githubusercontent.com/2042102/230414213-7ff22313-47c3-49ea-8b42-accb3c367a8c.png)

```
instrument.js:109 RangeError: Invalid time value
    at Date.toISOString (<anonymous>)
    at replayDataUtils.tsx:244:53
    at Array.map (<anonymous>)
    at f (replayDataUtils.tsx:231:6)
    at new d (replayReader.tsx:76:24)
    at d.factory (replayReader.tsx:48:12)
    at useReplayData.tsx:220:25
    at Object.zo [as useMemo] (react-dom.profiling.min.js:170:377)
    at o.useMemo (react.production.min.js:23:113)
    at m (useReplayData.tsx:219:18)
    at Ga (details.tsx:42:65)
    at uo (react-dom.profiling.min.js:159:137)
    at Yo (react-dom.profiling.min.js:183:154)
    at Ji (react-dom.profiling.min.js:278:449)
    at ju (react-dom.profiling.min.js:257:106)
    at Mu (react-dom.profiling.min.js:256:435)
    at Iu (react-dom.profiling.min.js:256:265)
    at Cu (react-dom.profiling.min.js:249:163)
    at react-dom.profiling.min.js:124:255
    at t.unstable_runWithPriority (scheduler.production.min.js:18:343)
    at Ga (react-dom.profiling.min.js:124:33)
    at Ya (react-dom.profiling.min.js:124:201)
    at Za (react-dom.profiling.min.js:124:136)
    at ku (react-dom.profiling.min.js:242:221)
    at Do (react-dom.profiling.min.js:172:408)
    at useReplayData.tsx:144:11
```

**Comment by JoshFerge** [comment]
thanks for reporting @alancnet. Confirming receipt & seing internally at https://sentry.sentry.io/issues/3919263890/ -- We will prioritize this ASAP.

**Comment by ryan953** [comment]
@alancnet We're still working on this.

TL/DR: It's taken longer than I expected, but we're overhauling the data types&parsing to fix this, and prevent this same kind of thing from happening again. 

I can add a little commentary and context too...

Another customer was seeing the same kind of `Invalid time value` error at this, and it turned out that they were setting a custom breadcrumb, but one of the timestamp fields was set to `undefined` unexpectedly. This is unhandled and caused the parsing error when the frontend reads the saved replay data. 

When I first started looking at the problem I found there are only few `toISOString()` calls, any of which could break in the same way. But then I looked more deeply at our types in the replay player code and found we're really loose about most fields. This was causing places to show `undefined` depending on the breadcrumb type, or other sporadic crashes depending on the use case. It's been a reason for lots of quick fixes over the last few months and a big drain on the team.

Where we are now:
We set out to overhaul the breadcrumb type system as it relates to replays, and we're in the last stretch of that effort. The idea is that we're going to have much simpler data hydration logic, and we're going to validate more fields, like this, upfront.

We've already caught a bug where the SDK said a new field is required, but it's added to an existing object: so therefore should be optional. So that's one less time "undefined" will show up on the screen.

**Comment by ryan953** [comment]
I pushed out some more quick fixes to get that example above rendering better, it's just a 5s replay but the errors, navigations and clicks are showing now. All we can do with the malformed custom breadcrumb is drop it i'm afraid.
